client/packages/system/src/Location/Components/LocationSearchInput.tsx

```diff
@@ -28,16 +70,21 @@ export const LocationSearchInput: FC<LocationSearchInputProps> = ({
   }, []);
 
   const locations = data?.nodes || [];
-  const options = [
-    ...locations.map(l => ({ value: l.id, label: l.name })),
-    { value: null, label: t('label.remove') },
-  ];
+  const options: AutocompleteOption<LocationOption>[] = locations.map(l => ({
+    value: l.id,
+    label: l.name,
+  }));
+
+  if (canRemove && locations.length > 0 && selectedLocation !== null) {
+    options.push({ value: null, label: t('label.remove') });
+  }
 
   return (
     <Autocomplete
       autoFocus={autoFocus}
       disabled={disabled}
       width={`${width}px`}
+      popperMinWidth={Number(width)}
```

Comment:
this one is to allow 'auto' width of the popper. Without it the popper will be fixed at 200px